### PR TITLE
Fix #116: Add RSS Feed

### LINKED ISSUE #116: RSS Feed for Blog
One of the things Gatsby does is it generates this in the `<head>`

```xml
<link rel="alternate" type="application/rss+xml" title="Melloware Blog" href="[/rss.xml]"/>
```

and example of the RSS feed it generates is

https://www.melloware.com/rss.xml

### PR DESCRIPTION
**Pull request by melloware**
Fix #116: Add RSS Feed

@ia3andy need a couple of things.

```xml
<description>
<![CDATA[ Melloware has been supporting the open source community since 1999. ]]>
</description>
<link>https://melloware.com</link>
<generator>GatsbyJS</generator>
<lastBuildDate>Tue, 01 Oct 2024 14:40:02 GMT</lastBuildDate>
```

The link `<link>https://melloware.com</link>` is `<link>/</link>` just a slash even though i need it to be the whole URL.

Other than that this all works!

here is what is being generated currently.

```xml
<rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
  <channel>
    <title><![CDATA[ Hello, world! I&#39;m Roq ]]></title>
    <description><![CDATA[ I provide you with all the tools to generate static websites out of your Quarkus web application. ]]></description>
    <link>/</link>
    <generator>Quarkus Roq</generator>
    <lastBuildDate>2024-10-09T12:00-04:00[America/New_York]</lastBuildDate>
      <item>
        <title><![CDATA[The second Roq plugin is for redirecting your page to a better place!]]></title>
        <link>/posts/2024-10-09-aliases</link>
        <guid isPermaLink="false">//posts/2024-10-09-aliases</guid>
        <pubDate>2024-10-09T12:00-04:00[America/New_York]</pubDate>
        <content:encoded><p>We introduced a way to declare aliases in FrontMatter. It is now easy create redirections to your blog posts!</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-10-09-aliases](view-source:http://posts/2024-10-09-aliases)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[The first Roq plugin is for tagging (with pagination)]]></title>
        <link>/posts/2024-10-08-tagging</link>
        <guid isPermaLink="false">//posts/2024-10-08-tagging</guid>
        <pubDate>2024-10-08T12:00-04:00[America/New_York]</pubDate>
        <content:encoded><p>We introduced the first Roq plugin, it is for collection tagging &amp; with pagination support!</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-10-08-tagging](view-source:http://posts/2024-10-08-tagging)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[Out of the box awesome SEO]]></title>
        <link>/posts/2024-09-23-seo</link>
        <guid isPermaLink="false">//posts/2024-09-23-seo</guid>
        <pubDate>2024-09-23T08:00-04:00[America/New_York]</pubDate>
        <content:encoded><p>Learn how to implement SEO in Roq in a blink of an eye.</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-09-23-seo](view-source:http://posts/2024-09-23-seo)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[Mastering Pagination in Roq]]></title>
        <link>/posts/2024-09-20-pagination</link>
        <guid isPermaLink="false">//posts/2024-09-20-pagination</guid>
        <pubDate>2024-09-20T08:00-04:00[America/New_York]</pubDate>
        <content:encoded><p>Learn how to implement pagination in Roq to enhance your content navigation. This article walks through the process of adding pagination, configuring page size, and customizing links.</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-09-20-pagination](view-source:http://posts/2024-09-20-pagination)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[How to add syntax highlighting to your Roq site]]></title>
        <link>/posts/2024-09-20-hightlight-js</link>
        <guid isPermaLink="false">//posts/2024-09-20-hightlight-js</guid>
        <pubDate>2024-09-20T05:00-04:00[America/New_York]</pubDate>
        <content:encoded><p>Learn how to integrate syntax highlighting into your Roq site using Highlight.js and the Quarkus web-bundler extension. This guide walks you through the simple steps to add it via the pom.xml, JavaScript, and SCSS files.</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-09-20-hightlight-js](view-source:http://posts/2024-09-20-hightlight-js)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[Easily manage Drafts and Future articles in Roq]]></title>
        <link>/posts/2024-09-19-drafts-and-future</link>
        <guid isPermaLink="false">//posts/2024-09-19-drafts-and-future</guid>
        <pubDate>2024-09-19T04:45-04:00[America/New_York]</pubDate>
        <content:encoded><p>Roq SSG introduces a new feature that allows you to hide or show draft and future articles using simple Quarkus configurations. This update gives developers greater control over which content is visible, improving content management and workflow.</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-09-19-drafts-and-future](view-source:http://posts/2024-09-19-drafts-and-future)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[Effortless URL Handling in Roq with Qute super-power]]></title>
        <link>/posts/2024-09-16-nice-url-experience</link>
        <guid isPermaLink="false">//posts/2024-09-16-nice-url-experience</guid>
        <pubDate>2024-09-16T07:32:20-04:00[America/New_York]</pubDate>
        <content:encoded><p>Effortlessly manage both relative and absolute URLs with our enhanced Qute-powered feature. Utilizing the RoqUrl class, you can easily join and resolve paths, ensuring clean and predictable URLs. This update simplifies URL handling, making your code more efficient and your content easier to navigate and share.</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-09-16-nice-url-experience](view-source:http://posts/2024-09-16-nice-url-experience)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
      <item>
        <title><![CDATA[Welcome to Roq!]]></title>
        <link>/posts/2024-08-29-welcome-to-roq</link>
        <guid isPermaLink="false">//posts/2024-08-29-welcome-to-roq</guid>
        <pubDate>2024-08-29T07:32:20-04:00[America/New_York]</pubDate>
        <content:encoded><p>This is the first article ever made with Quarkus Roq</p><div style="margin-top: 50px; font-style: italic;"><strong><a href="[//posts/2024-08-29-welcome-to-roq](view-source:http://posts/2024-08-29-welcome-to-roq)">Keep reading</a>.</strong></div><br /> <br /></content:encoded>
      </item>
  </channel>
</rss>

```

### CODE REVIEW

File: roq-frontmatter/runtime/src/main/resources/templates/fm/rss.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{@io.quarkiverse.roq.frontmatter.runtime.model.Site site}
+{@io.quarkiverse.roq.frontmatter.runtime.model.NormalPage page}
+
+<rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
+  <channel>
+    <title><![CDATA[ {site.title} ]]></title>
+    <description><![CDATA[ {site.description} ]]></description>
+    <link>{site.url}</link>
```
Comment (ia3andy) [inline roq-frontmatter/runtime/src/main/resources/templates/fm/rss.html]: `site.url.absolute`

File: roq-frontmatter/runtime/src/main/resources/templates/fm/rss.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <generator>Quarkus Roq</generator>
+    <lastBuildDate>{#if site.collections.posts.size > 0}{site.collections.posts[0].date}{/if}</lastBuildDate>
+    {#for post in site.collections.posts}
+      <item>
+        <title><![CDATA[{post.title}]]></title>
+        <link>{site.url}{post.url}</link>
```
Comment (ia3andy) [inline roq-frontmatter/runtime/src/main/resources/templates/fm/rss.html]:
```suggestion
        <link>{post.url.absolute}</link>
```

File: roq-frontmatter/runtime/src/main/resources/templates/fm/rss.html (hunk 1 above)
Comment (ia3andy) [inline roq-frontmatter/runtime/src/main/resources/templates/fm/rss.html]:
```suggestion
    <link>{site.url.absolute}</link>
```

File: blog/site/index.html [hunk 3]
```diff
@@ -9,6 +9,7 @@
 social-github: quarkiverse/quarkus-roq
 social-linkedin: quarkusio
 layout: main
+url: https://pages.quarkiverse.io/quarkus-roq/
```
Comment (ia3andy) [inline blog/site/index.html]:
```suggestion
```
Not needed

**Review by ia3andy — APPROVED** [review]
After fixing the FM you can merge

**Comment by ia3andy** [comment]
Thanks!!!

**Comment by melloware** [comment]
Will this auto deploy i would like to inspect it on the running GitHub blog when it deploys?

**Comment by ia3andy** [comment]
https://pages.quarkiverse.io/quarkus-roq/rss.xml/

**Pull request merged**